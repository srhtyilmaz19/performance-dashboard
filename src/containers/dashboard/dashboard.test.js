import React from "react";
import { cleanup, render, screen } from "@testing-library/react";
import { combineReducers, createStore } from "redux";
import { Provider } from "react-redux";
import "@testing-library/jest-dom/extend-expect";

import Dashboard from "./dashboard";
import types from "./action-types";

afterEach(cleanup);

const initialState = {
  loading: true,
  error: false,
  metrics: [],
  charts: [
    { key: "fcp", title: "First Contentful Paint", data: [] },
    { key: "ttfb", title: "Time To First Byte", data: [] },
    { key: "dom_load", title: "DOM Load", data: [] },
    { key: "window_load", title: "Window Load", data: [] },
  ],
};

function reducer(state = initialState, action) {
  switch (action.type) {
    case types.GET_DOMAIN_METRICS:
      return {
        ...state,
        loading: true,
        error: false,
        metrics: [],
      };

    case types.GET_DOMAIN_METRICS_SUCCESS:
      console.log(action.data);
      return {
        ...state,
        metrics: action.data.metrics,
        loading: false,
      };

    case types.GET_DOMAIN_METRICS_ERROR:
      return {
        ...state,
        metrics: [],
        loading: false,
        error: true,
      };

    default:
      return state;
  }
}

const combinedReducers = combineReducers({
  dashboard: reducer,
});

function renderWithRedux(Component) {
  return {
    ...render(
      <Provider store={createStore(combinedReducers)}> {Component} </Provider>
    ),
  };
}

describe("<Dashboard/>", () => {
  it("should contain 2 action-buttons", async () => {
    renderWithRedux(<Dashboard />);

    const buttons = screen.getAllByRole("button");

    expect(Object.keys(buttons).length).toBe(2);
  });

  it("should contain 2 date-time-picker inputs", async () => {
    renderWithRedux(<Dashboard />);

    const buttons = screen.getAllByRole("input", { type: "text" });

    expect(Object.keys(buttons).length).toBe(2);
  });
});
