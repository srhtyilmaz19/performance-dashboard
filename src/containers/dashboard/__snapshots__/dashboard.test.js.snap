// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard/> should render with redux 1`] = `
<DocumentFragment>
   
  <div
    class="dashboard-wrapper"
  >
    <h2>
      Performance Metrics !
    </h2>
    <div
      class="date-picker-wrapper"
    >
      <div
        class="makeStyles-container-1"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-textField-2"
          role="input"
        >
          <input
            class="MuiInputBase-input MuiInput-input"
            id="start_date"
            type="text"
            value="2021/05/08 23:11"
          />
        </div>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-textField-2"
          role="input"
        >
          <input
            class="MuiInputBase-input MuiInput-input"
            id="end_date"
            type="text"
            value="2021/05/08 23:41"
          />
        </div>
      </div>
      <button
        class="button"
        data-testid="filter-button"
        type="button"
      >
        FILTER
      </button>
      <button
        class="button"
        data-testid="reset-button"
        type="button"
      >
        RESET
      </button>
    </div>
    <div
      class="loading-wrapper"
      data-testid="loading-wrapper"
    >
      <div
        class="lds-dual-ring"
        data-testid="loading-inner"
      />
      <span
        data-testid="loading-text"
      >
        Loading.
      </span>
    </div>
  </div>
   
</DocumentFragment>
`;
